### DISCUSSION TITLE
High disk usage when doing RTSP to WebRTC

### DISCUSSION THREAD

**Discussion by mariia-artamonova** [ant-media/Ant-Media-Server#5949] [FAQs] [answered]
Hi! We have 5 live streams running with RTSP to WebRTC, and sometimes, just sometimes, disk usage goes up by multiple gigabytes each hour. Recording is not enabled.

To fix this, we needed to stop all the streams, and manually delete all transport stream files. (We saw up to 700k of these files in the WebRTC/streams folder)

![High disk usage when doing RTSP to WebRTC on ant media server](https://github.com/ant-media/Ant-Media-Server/assets/149576688/60340d83-22b6-4f06-a548-0e312ef6d02c)

One other problem we are seeing; If a recording is running and the disk fills completely up, the AMS stops and refuses to start again until we manually delete the VODs from the server.

Are there anyone else that has encountered the same things, and know a fix for it?

**Comment by yashtandon113** [comment, marked as answer]
@mariia-artamonova 
Thanks for your questions. Let me try to answer.

1. It seems all the HLS files are stored on the server. It was a known bug in v2.4.3. Please check out the below link for an explanation of this issue and a workaround solution.
https://github.com/ant-media/Ant-Media-Server/issues/4196

2. Yes, it is an expected behavior. If the recording is ongoing, the tmp extension file will take up space on the server's local disk until the recording is completed.

- For the solution to this problem, you may mount an extra storage device in the streams directory of the Ant Media server.

- Another option is to use S3 Fuse to map your stream directory to cloud storage (an S3 bucket). Kindly check out [this discussion](https://github.com/orgs/ant-media/discussions/5090) for multiple options to store MP4 or HLS on the bucket.

- Also check out the [new feature](https://antmedia.io/docs/guides/recording-live-streams/s3-integration-http-forwarding/#uploading-hls-files-to-the-s3-bucket-in-real-time) to upload HLS stream to the bucket on the fly without filling up the disk space.

I hope this helps.
